### DISCUSSION TITLE
Integration of C/C++ repositories

### DISCUSSION THREAD

**Discussion by Trakeur** [DependencyTrack/dependency-track#2801] [Ideas]
Hi everyone,

I was wondering why there was no repository for the C/C++ language in dependency-track? (With Conan for example)
I guess this has already been addressed but I haven't found any related discussions.

Is it easy to integrate?

Thank you.

**Comment by syalioune** [comment]
Hello @Trakeur 

> I was wondering why there was no repository for the C/C++ language in dependency-track? (With Conan for example)

I guess it's just a time and community priority matter. Since the project is fueled by a small number of people, there's only so much that can be done. Contributions are always welcomed !

If you have Java and Vue.js knowledge, you can take a look at the recent implementation for Perl/CPAN in the [backend](https://github.com/DependencyTrack/dependency-track/pull/2593/files) and [frontend](https://github.com/DependencyTrack/frontend/pull/441/files). It's quite simple actually, you have to provide a way to map a [Conan purl](https://github.com/package-url/purl-spec/blob/master/PURL-TYPES.rst#conan) to a REST API call to the repository in order to fetch package metadata, and provide the logic deciding if the current version is the latest one based on the retrieved metadata.

**Reply by Trakeur** [reply]
Hi @syalioune,

Thanks for your reply, I understand, I imagine it wasn't the most popular language at first. I do have some knowledge of java but not at all of Vue.js, so I could try to do a few things on the backend.

Given the very large number of repositories using PURL to identify their components, I was also wondering if there had ever been an attempt to offer the possibility of adding any ecosystem from the web interface, for flexibility on any project?

**Reply by syalioune** [reply]
I don't think it is possible to integrate a new ecosystem by pure configuration because even though the principle is constant, the repositories API /model/logic are all differents

**Reply by Trakeur** [reply]
Yes, it's true, at least it's probably feasible, but it would require a lot of manual configuration.
